x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/edit_role_mapping_page.tsx

```diff
@@ -211,6 +214,49 @@ export class EditRoleMappingPage extends Component<Props, State> {
     );
   };
 
+  private checkEmptyAnyAllMappings = (obj: RoleMappingRule) => {
+    const arrToCheck: RoleMappingRule[] = [obj];
+
+    while (arrToCheck.length > 0) {
+      const currentObj = arrToCheck.pop();
+      if (typeof currentObj === 'object' && currentObj !== null) {
+        for (const key in currentObj) {
+          if (currentObj.hasOwnProperty(key)) {
+            const value: unknown = currentObj[key as keyof RoleMappingRule];
+            if ((key === 'any' || key === 'all') && Array.isArray(value)) {
+              if (value.length === 0) {
+                return true;
+              }
+              arrToCheck.push(...value);
+            } else if (typeof value === 'object' && value !== null) {
+              arrToCheck.push(value as RoleMappingRule);
+            }
+          }
+        }
+      }
+    }
+    return false;
+  };
+
+  private getFormWarnings = () => {
+    if (this.checkEmptyAnyAllMappings(this.state.roleMapping!.rules as RoleMappingRule)) {
+      return (
+        <EuiCallOut
+          title="Warning"
+          color="warning"
+          iconType="alert"
+          data-test-subj="emptyAnyOrAllRulesWarning"
+        >
+          <FormattedMessage
+            id="xpack.security.management.editRoleMapping.emptyAnyAllMappingsWarning"
+            defaultMessage="Role mapping rules contains empty 'any' or 'all' rules. Please proceed with caution"
```

Comment:
This message states the "what", but not the "why". I recommend including language that educates the user, by explaining that these empty rule groups will always evaluate to `true`. We can't expect them to already know this behavior, and likely won't know why they are being asked to proceed with caution.

cc @elastic/kibana-docs can help with wording if necessary.